DOC: Spellcheck of gotchas.rst (FAQ page)

tommyod opened the pull request:
Minor changes to the documentation, specifically `gotchas.rst`, i.e. the FAQ page.

* Function/method references as links.
* Consistent double backticks ` `` ` around Series, DataFrame.
* Minor rephrasing of a sentence or two, added some missing punctuation.
* Replaced `.. ipython::` with `ipython:: python` twice, as the former is not sufficient to make the iPython code render in the docs.

Reviews/comments very welcome.

tommyod committed:
Spellchecked gotchas.rst

tommyod committed:
Fixed some additional, minor spelling mistakes

jreback commented:
> Replaced .. ipython:: with ipython:: python twice, as the former is not sufficient to make the iPython code render in the docs.

would welcome this an additional check in the lint.sh

jreback commented on doc/source/gotchas.rst:
```diff
@@ -22,21 +22,20 @@ Frequently Asked Questions (FAQ)
     n = 5000
-    data = dict([ (t, np.random.randint(100, size=n).astype(t))
+    data = dict([(t, np.random.randint(100, size=n).astype(t))
                     for t in dtypes])
```
can you fix indent here

jreback commented on doc/source/gotchas.rst:
```diff
@@ -22,21 +22,20 @@ Frequently Asked Questions (FAQ)
-specifies if the dataframe's memory usage will be displayed when
-invoking the ``df.info()`` method.
+The memory usage of a ``DataFrame`` (including the index) is shown when calling
+the :meth:`~DataFrame.info`. A configuration option, ``display.memory_usage`` 
```
can you reference this option directly?

jreback commented on doc/source/gotchas.rst:
```diff
@@ -320,7 +323,7 @@ a different byte order than the one on which you are running Python. A common sy
 To deal
 with this issue you should convert the underlying NumPy array to the native
-system byte order *before* passing it to Series/DataFrame/Panel constructors
+system byte order *before* passing it to ``Series``/``DataFrame`` constructors
```
/ -> or

tommyod committed:
Adressed comments. Added linting rule.

tommyod committed:
Fixed indentation in lint.sh

jorisvandenbossche commented:
There are certain legitimate use cases of `.. ipython::` blocks with the `python` added, so I am not sure we can add such a lint check in general.

jorisvandenbossche commented:
For example, the code snippet here is done with a bar `.. ipython::` on purpose: http://pandas.pydata.org/pandas-docs/stable/basics.html#iteritems. I don't recall exactly, but I think it was a problem with the for loop in a ipython::python directive (but it might be that that works correctly by now, but we would need to check).

jreback commented on ci/lint.sh:
```diff
@@ -149,6 +149,15 @@ if [ "$LINT" ]; then
         RET=1
     fi
     echo "Check for deprecated messages without sphinx directive DONE"
+
```
does this pass now?

jreback commented:
small question. looks fine. can you rebase and push again just to make sure.

jorisvandenbossche commented:
@jreback no, this is not ready, see my comment

BTW, can you answer https://github.com/pandas-dev/pandas/issues/19750 before merging this?

jorisvandenbossche commented:
Ah, I didn't see that the CI was passing. But since we still have occurrences of `.. ipython::` without `python` in our docs, that means the lint is check is actually not working?

tommyod committed:
Removed lint for '.. python:' without python

tommyod commented:
I believe this PR is good to go now.

Thanks for double-checking the lint @jorisvandenbossche. It seems like it was not working, glad you caught it. Since there might be cases which warrant `.. ipython::` without `python`, I have removed the lint for now. In the particular case of the code blocks where I added `python` in this PR, the code is [currently not rendered](http://pandas-docs.github.io/pandas-docs-travis/gotchas.html#using-the-in-operator) - so adding `python` seems correct as it now renders on my local doc build.

jreback commented:
``.. ipython:: without python``

I guess, but why can't we cross this bridge i its actually needed. not-specifying python makes this look not so great and easy enough to avoid with a lint check.

jorisvandenbossche commented:
The cases of `.. ipython:: without python` that @tommyod fixed, where indeed correct fixes (it were cases where we forgot this, and by forgetting it indeed did not render correctly in the docs). 

But as I mentioned above, there are *other* cases where this was done on purpose (https://github.com/pandas-dev/pandas/pull/19747#issuecomment-366547716). And in those they are currently rendered fine. I am not sure if those cases are actually needed, but if someone wants to remove them, we should first check the docs build correctly without them.

jreback commented:
ok that’s fine 
better to have automatically checks in things if possible otherwise end up with inconsistencies over time
